File: src/main/java/kr/co/dealmungchi/hotdealbatch/dto/HotDealDto.java
package kr.co.dealmungchi.hotdealbatch.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public record HotDealDto(
    String title,
    String link,
    String price,
    String thumbnail,
    @JsonProperty("thumbnail_link")
    String thumbnailLink,
    
    @JsonProperty("id")
    String postId,
    
    @JsonProperty("posted_at")
    String postedAt,
    
    String provider
) {}

File: src/main/kotlin/kr/co/dealmungchi/hotdealbatch/dto/HotDealDto.kt
package kr.co.dealmungchi.hotdealbatch.dto

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Data transfer object for hot deals.
 */
data class HotDealDto(
    val title: String,
    val link: String,
    val price: String?,
    val thumbnail: String?,
    
    @JsonProperty("thumbnail_link")
    val thumbnailLink: String?,
    
    @JsonProperty("id")
    val postId: String,
    
    @JsonProperty("posted_at")
    val postedAt: String?,

    @JsonProperty("category")
    val category: String?,
    
    val provider: String
)